<?xml version="1.0"?>
<launch>


    <arg name="path_to_orb" default="/home/usrg/packages/ORB_SLAM3_ROS_Interface/"/>
    <arg name="path_to_vocabulary" default="Vocabulary/ORBvoc.txt"/>
    <arg name="path_to_settings" default="Examples/RGB-D/nn_conversion.yaml"/>

    <arg name="use_pangolin" default="false"/>
    <arg name="use_auto_exposure" default="true"/>
    <arg name="ae_meanIntensitySetPoint" default="1500"/>

    <!-- conversion node -->
    <node name="mono_sub_topic_conversion" pkg="orb3_ros_interface" type="mono_sub_topic" output="screen"
    args="$(arg path_to_orb)$(arg path_to_vocabulary) $(arg path_to_orb)$(arg path_to_settings)" >
      <remap from="/orb_pose" to="/scout/mavros/vision_pose/pose"/>
      <remap from="/camera/aligned_depth_to_color/image_raw" to="/camera/depth/image_rect_raw"/>
      <param name="enable_pangolin" type="bool" value="$(arg use_pangolin)" />
      <param name="enable_auto_exposure" type="bool" value="$(arg use_auto_exposure)" />
      <param name="ae_meanIntensitySetPoint" type="int" value="$(arg ae_meanIntensitySetPoint)" />

      <remap from="/camera/color/image_raw" to="/camera/image_conversed"/> 
    </node>

    <!-- originanl node -->
    <!-- <node name="mono_sub_topic" pkg="orb3_ros_interface" type="mono_sub_topic" output="screen"
    args="$(arg path_to_orb)$(arg path_to_vocabulary) $(arg path_to_orb)$(arg path_to_settings)" >
      <remap from="/orb_pose" to="/scout/mavros/vision_pose/pose"/>
      <remap from="/camera/aligned_depth_to_color/image_raw" to="/camera/depth/image_rect_raw"/>
      <param name="enable_pangolin" type="bool" value="$(arg use_pangolin)" />
      <param name="enable_auto_exposure" type="bool" value="$(arg use_auto_exposure)" />
      <param name="ae_meanIntensitySetPoint" type="int" value="$(arg ae_meanIntensitySetPoint)" />

      
      <remap from="/orb3_feature_image_from_orb" to="/orb3_feature_image_original"/> 
    </node> -->

    <!--<remap from="/camera/color/image_raw" to="/camera/image_conversed"/> -->
</launch>
